src/engraving/libmscore/cmd.cpp

```diff
@@ -331,6 +331,24 @@ void Score::undoRedo(bool undo, EditData* ed)
     }
 }
 
+void Score::resetAndClearUndoStack(EditData* ed)
+{
+    if (readOnly()) {
+        return;
+    }
+
+    cmdState().reset();
+
+    auto undoStack = this->undoStack();
+    while (!undoStack->isClean()) {
+        undoStack->undo(ed);
```

Comment:
This seems to me a very risky way of reverting to the last saved state. If the score has become corrupted, I would trust the UndoStack even less. In some cases with corruptions or crashes that I've worked on, it looked indeed like undoing only made it worse. 
I thought we would just close and completely re-read the file from disk, to remove any risk.

Reply:
yep, fixed